libs/actions/sendmail.c

```diff
@@ -0,0 +1,226 @@
+*/
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <fcntl.h>
+
+#include "action.h"
+#include "log.h"
+#include "dso.h"
+#include "json.h"
+#include "common.h"
+#include "config.h"
+#include "pilight.h"
+#include "mail.h"
+#include "sendmail.h"
+#include "settings.h"
+
+
+//check arguments and settings
+static int actionSendmailArguments(struct JsonNode *arguments) {
+	struct JsonNode *jsubject = NULL;
+	struct JsonNode *jmessage = NULL;
+	struct JsonNode *jto = NULL;
+	struct JsonNode *jvalues = NULL;
+	struct JsonNode *jchild = NULL;
+	struct JsonNode *temp = NULL;
+	char *stmp = NULL;
+	int nrvalues = 0, itmp = 0;
+	
+	jsubject = json_find_member(arguments, "SUBJECT");
+	jmessage = json_find_member(arguments, "MESSAGE");
+	jto = json_find_member(arguments, "TO");
+	
+	if(jmessage == NULL) {
+		logprintf(LOG_ERR, "sendmail action is missing a \"MESSAGE\"");
+		return -1;
+	}
+	nrvalues = 0;
+	if((jvalues = json_find_member(jsubject, "value")) != NULL) {
+		jchild = json_first_child(jvalues);
+		while(jchild) {
+			nrvalues++;
+			jchild = jchild->next;
+		}
+	}
+	if(nrvalues > 1) {
+		logprintf(LOG_ERR, "sendmail action \"SUBJECT\" only takes one argument");
+		return -1;
+	}
+	nrvalues = 0;
+	if((jvalues = json_find_member(jmessage, "value")) != NULL) {
+		jchild = json_first_child(jvalues);
+		while(jchild) {
+			nrvalues++;
+			jchild = jchild->next;
+		}
+	}
+	if(nrvalues != 1) {
+		logprintf(LOG_ERR, "sendmail action \"MESSAGE\" only takes one argument");
+		return -1;
+	}
+	nrvalues = 0;
+	if((jto = json_find_member(jmessage, "value")) != NULL) {
+		jchild = json_first_child(jvalues);
+		while(jchild) {
+			nrvalues++;
+			jchild = jchild->next;
+		}
+	}
+	if(nrvalues > 1) {
+		logprintf(LOG_ERR, "sendmail action \"TO\" only takes one argument");
+		return -1;
+	}
+	// Check if mandatory settings are present in config
+	if(settings_find_string("smtp-host", &stmp) != EXIT_SUCCESS) {
+		logprintf(LOG_ERR, "Sendmail: setting \"smtp-host\" is missing in config");
+		return -1;
+	} 
+	if(settings_find_number("smtp-port", &itmp) != EXIT_SUCCESS) {
+		logprintf(LOG_ERR, "Sendmail: setting \"smtp-port\" is missing in config");
+		return -1;
+	} 
+	if(settings_find_string("smtp-user", &stmp) != EXIT_SUCCESS) {
+		logprintf(LOG_ERR, "Sendmail: setting \"smtp-user\" is missing in config");
+		return -1;
+	} 
+	if(settings_find_string("smtp-password", &stmp) != EXIT_SUCCESS) {
+		logprintf(LOG_ERR, "Sendmail: setting \"smtp-password\" is missing in config");
+		return -1;
+	} 
+	if(settings_find_string("mail-to", &stmp) != EXIT_SUCCESS) {
+		logprintf(LOG_ERR, "Sendmail: setting \"mail-to\" is missing in config");
+		return -1;
+	} 
+	if(settings_find_string("mail-from", &stmp) != EXIT_SUCCESS) {
+		logprintf(LOG_ERR, "Sendmail: setting \"mail-from\" is missing in config");
+		return -1;
+	} 
+	if(settings_find_string("mail-subject", &stmp) != EXIT_SUCCESS) {
```

Comment:
The `SUBJECT` is also part of the mail action arguments, so why do we have a `mail-subject` setting?